grails-app/domain/org/pih/warehouse/order/Order.groovy

```diff
@@ -518,6 +518,11 @@ class Order implements Serializable {
         return orderType?.isTransferOrder()
     }
 
+    Boolean isFullyInvoiceable() {
+        Set encumberedOrderItems = orderItems.findAll{ it -> it.encumbered }
```

Comment:
why not specifing the type of the Set? `Set<OrderItem>`?

Reply:
```suggestion
        Set<OrderItem> encumberedOrderItems = orderItems.findAll{ it -> it.encumbered }
```

Reply:
and one more question - does it parse it to the Set correctly? In general the `.findAll` returns `List<T>`

Reply:
The IDE suggestion was that it returned a `Set<OrderItems>`

Reply:
Any particular reason to not do it in one loop with something like:
`return !orderItems.any{ it -> it.encumbered && !it.invoiceable }`